orca-clouddriver/src/test/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/DetermineHealthProvidersTaskSpec.groovy

```diff
@@ -65,6 +76,38 @@ class DetermineHealthProvidersTaskSpec extends Specification {
     [application: "app"]                          | null                   || [:]                                           // no health provider names when an exception is raised
   }
 
+  @Unroll
+  def "should check interestingHealthProviderNames based server group 'interestingHealthProviderNames' label for titus "() {
+    given:
+    def stage = new PipelineStage(new Pipeline(), stageType, stageContext)
+    def response = mapper.writeValueAsString([
+      application: "app",
+      region: "region",
+      account: "test",
+      labels: labels
+    ])
+
+    and:
+    1 * front50Service.get("app") >> bA("app", false, false)
+    invocationCount * oortService.getServerGroup(*_) >> new Response('oort', 200, 'ok', [], new TypedString(response))
+
+    when:
+    def taskResult = task.execute(stage)
+
+    then:
+    taskResult.status == ExecutionStatus.SUCCEEDED
+    (taskResult.getStageOutputs() as Map) == expectedStageOutputs
+
+    where:
+    stageType             | stageContext                                                           | labels                                                  | invocationCount || expectedStageOutputs
+    "createServerGroup"   | [application: "app", strategy: "rollingpush", cloudProvider: "titus"]  | [interestingHealthProviderNames: "Titus"]               | 1               || [interestingHealthProviderNames: ["Titus"]]
+    "createServerGroup"   | [application: "app", strategy: "rollingpush", cloudProvider: "aws"]    | [interestingHealthProviderNames: "Titus"]               | 0               || [:]
+    "createServerGroup"   | [application: "app", strategy: "rollingpush", cloudProvider: "titus"]  | [interestingHealthProviderNames: "Titus,Discovery"]     | 1               || [interestingHealthProviderNames: ["Titus", "Discovery"]]
+    "createServerGroup"   | [application: "app", strategy: "rollingpush", cloudProvider: "titus"]  | null                                                    | 1               || [interestingHealthProviderNames: []]
```

Comment:
In this case, it seems more readable to only set it if that collection isn't empty.